common/src/main/java/com/wynntils/features/SoulPointTimerFeature.java

```diff
@@ -17,61 +15,48 @@
 import net.minecraftforge.eventbus.api.SubscribeEvent;
 
 @FeatureInfo(stability = Stability.STABLE, gameplay = GameplayImpact.MEDIUM, performance = PerformanceImpact.MEDIUM)
-public class SoulPointTimerFeature extends Feature {
-    @Override
-    protected void onInit(ImmutableList.Builder<Condition> conditions) {}
+public class SoulPointTimerFeature extends FeatureBase {
 
-    @Override
-    protected boolean onEnable() {
-        WynntilsMod.getEventBus().register(this);
-        return true;
-    }
-
-    @Override
-    protected void onDisable() {
-        WynntilsMod.getEventBus().unregister(this);
-    }
-
-    @Override
-    public MutableComponent getNameComponent() {
-        return new TranslatableComponent("feature.wynntils.soulPointTimer.name");
+    public SoulPointTimerFeature() {
+        setupEventListener();
     }
 
     @SubscribeEvent
     public void onItemsReceived(ItemsReceivedEvent e) {
         if (!WynnUtils.onServer()) return;
 
-        for (ItemStack stack : e.getItems()) {
-            if (!WynnItemMatchers.isSoulPoint(stack)) continue;
+        ItemStack soulPointStack = getSoulPointStack(e.getItems());
+        if (soulPointStack == null) return;
 
-            ListTag lore = ItemUtils.getLoreTag(stack);
+        ListTag lore = ItemUtils.getLoreTag(soulPointStack);
 
-            if (lore == null) {
-                lore = new ListTag();
-            } else {
-                lore.add(StringTag.valueOf("")); // Equivalent to adding ""
-            }
+        if (lore == null) {
+            lore = new ListTag();
+        } else {
+            lore.add(StringTag.valueOf("")); // Equivalent to adding ""
+        }
 
-            lore.add(new DynamicTag(() -> {
-                int rawSecondsUntilSoulPoint = WynnInventoryData.getTicksTillNextSoulPoint() / 20;
-                int minutesUntilSoulPoint = rawSecondsUntilSoulPoint / 60;
-                int secondsUntilSoulPoint = rawSecondsUntilSoulPoint % 60;
+        lore.add(new DynamicTag(() -> {
+            int rawSecondsUntilSoulPoint = WynnInventoryData.getTicksTillNextSoulPoint() / 20;
+            int minutesUntilSoulPoint = rawSecondsUntilSoulPoint / 60;
+            int secondsUntilSoulPoint = rawSecondsUntilSoulPoint % 60;
 
-                return ItemUtils.toLoreString(
-                        ChatFormatting.AQUA + "Time until next soul point: " + ChatFormatting.WHITE
-                                + minutesUntilSoulPoint + ":" + String.format("%02d", secondsUntilSoulPoint));
-            }));
+            return ItemUtils.toLoreString(ChatFormatting.AQUA + "Time until next soul point: " + ChatFormatting.WHITE
+                    + minutesUntilSoulPoint + ":" + String.format("%02d", secondsUntilSoulPoint));
+        }));
 
-            ItemUtils.replaceLore(stack, lore);
-        }
+        ItemUtils.replaceLore(soulPointStack, lore);
+    }
+
+    private static ItemStack getSoulPointStack(List<ItemStack> items) {
```

Comment:
might be nicer to name it `findSoulPointStack`

Reply:
Are we not actually guaranteed to find it? I mean, if we were actually checking if we had soul points, I'd understand this name. But now it's more about locating the proper spot where the soul points are stored, right?

Reply:
yeah

Reply:
Ok fixed.